overlord/snapstate/conflict.go

```diff
@@ -113,25 +114,27 @@ func snapSetupFromChange(chg *state.Change) (*SnapSetup, error) {
 // for a snapd refresh/revert. While snapd is updating we do not allow other
 // snap changes to be executing, to avoid any incompatibilities in the task system,
 // between snapd versions.
-func changeIsSnapdDowngrade(st *state.State, chg *state.Change) bool {
+func changeIsSnapdDowngrade(st *state.State, chg *state.Change) (bool, error) {
 	snapsup, err := snapSetupFromChange(chg)
 	if err != nil || snapsup == nil || snapsup.SnapName() != "snapd" {
-		return false
+		return false, nil
 	}
 
 	var snapst SnapState
-	err = Get(st, snapsup.InstanceName(), &snapst)
+	if err := Get(st, snapsup.InstanceName(), &snapst); err != nil {
+		return false, err
+	}
+
+	currentInfo, err := snapst.CurrentInfo()
 	if err != nil {
-		return false
+		return false, fmt.Errorf("cannot retrieve snap info for current snapd: %v", err)
 	}
 
-	// If snapd was not installed, or if the revision we are installing
-	// is local do not consider this a downgrade. (If the revision currently
-	// install is local then the check below will return false anyway).
-	if !snapst.IsInstalled() || snapsup.Revision().Local() {
-		return false
+	res, err := strutil.VersionCompare(currentInfo.Version, snapsup.Version)
```

Comment:
snapsup.Version could be "" in case the change originates with an older snapd, in which case I suppose we assume it's a downgrade as we don't know better